CMakeLists.txt

```diff
@@ -1,111 +1,104 @@
+             OUTPUT_NAME example
 )
 
 target_include_directories(
-    example_example
-    ${example_warning_guard}
-    PUBLIC
-    "$<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include>"
-    "$<BUILD_INTERFACE:${PROJECT_BINARY_DIR}/include>"
+  example_example # XXX ${example_warning_guard}
+  PUBLIC "$<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include>"
+         "$<BUILD_INTERFACE:${PROJECT_BINARY_DIR}/include>"
+         "$<INSTALL_INTERFACE:${EXAMPLE_INCLUDE_DIRECTORY}>"
 )
 
-target_compile_features(example_example PUBLIC cxx_std_14)
+target_compile_features(example_example PUBLIC cxx_std_17)
 
 # ---- Install ----
 
 include(CMakePackageConfigHelpers)
-include(GNUInstallDirs)
-
-set(example_include_directory "${CMAKE_INSTALL_INCLUDEDIR}/example")
 
 install(
-    DIRECTORY
-    "${PROJECT_SOURCE_DIR}/include/"
-    "${PROJECT_BINARY_DIR}/include/"
-    DESTINATION "${example_include_directory}"
-    COMPONENT example_Development
+  DIRECTORY "${PROJECT_SOURCE_DIR}/include/" "${PROJECT_BINARY_DIR}/include/"
+  DESTINATION "${EXAMPLE_INCLUDE_DIRECTORY}"
+  COMPONENT example_Development
 )
 
 install(
-    TARGETS example_example
-    EXPORT exampleTargets
-    RUNTIME #
-    COMPONENT example_Runtime
-    LIBRARY #
-    COMPONENT example_Runtime
-    NAMELINK_COMPONENT example_Development
-    ARCHIVE #
-    COMPONENT example_Development
-    INCLUDES #
-    DESTINATION "${example_include_directory}"
+  TARGETS example_example
+  EXPORT exampleTargets
+  RUNTIME COMPONENT example_Runtime
+  LIBRARY COMPONENT example_Runtime NAMELINK_COMPONENT example_Development
+  ARCHIVE COMPONENT example_Development
+  INCLUDES
+  DESTINATION "${EXAMPLE_INCLUDE_DIRECTORY}"
 )
 
 write_basic_package_version_file(
-    exampleConfigVersion.cmake
-    COMPATIBILITY SameMajorVersion
+  exampleConfigVersion.cmake COMPATIBILITY SameMajorVersion
 )
 
-set(
-    example_INSTALL_CMAKEDIR "${CMAKE_INSTALL_LIBDIR}/cmake/example"
+set(EXAMPLE_INSTALL_CMAKEDIR
+    "${CMAKE_INSTALL_LIBDIR}/cmake/example"
     CACHE STRING "CMake package config location relative to the install prefix"
 )
-mark_as_advanced(example_INSTALL_CMAKEDIR)
+mark_as_advanced(EXAMPLE_INSTALL_CMAKEDIR)
 
 install(
-    FILES
-    "${PROJECT_SOURCE_DIR}/cmake/exampleConfig.cmake"
-    "${PROJECT_BINARY_DIR}/exampleConfigVersion.cmake"
-    DESTINATION "${example_INSTALL_CMAKEDIR}"
-    COMPONENT example_Development
+  FILES "${PROJECT_SOURCE_DIR}/cmake/exampleConfig.cmake"
+        "${PROJECT_BINARY_DIR}/exampleConfigVersion.cmake"
+  DESTINATION "${EXAMPLE_INSTALL_CMAKEDIR}"
+  COMPONENT example_Development
 )
 
 install(
-    EXPORT exampleTargets
-    NAMESPACE example::
-    DESTINATION "${example_INSTALL_CMAKEDIR}"
-    COMPONENT example_Development
+  EXPORT exampleTargets
+  NAMESPACE example::
+  DESTINATION "${EXAMPLE_INSTALL_CMAKEDIR}"
+  COMPONENT example_Development
 )
 
-if(CMAKE_SOURCE_DIR STREQUAL CMAKE_CURRENT_SOURCE_DIR)
+if("${CMAKE_SOURCE_DIR}" STREQUAL "${CMAKE_CURRENT_SOURCE_DIR}")
```

Comment:
This is a pointless change. Please learn how the `if` command works: https://cmake.org/cmake/help/latest/command/if.html

Reply:
if(<variable|string> STREQUAL <variable|string>)
True if the given string or variable's value is lexicographically equal to the string or variable on the right.

I know, but everything in CMAKE is first a string!
is seems more clear to be explicit.

Reply:
`if()`'s documentation is already explicit enough about what it does.